packages/design-system/src/components/web-components/ds-tabs/ds-tabs.tsx

```diff
@@ -1,20 +1,44 @@
 import { define } from '../preactement/define';
 import { Tabs, TabPanel } from '../../Tabs';
+import { TabPanelProps } from '../../Tabs/TabPanel';
 import { findElementsOfType } from '../../utilities/findElementsOfType';
 import { createElement } from 'react';
 
-const attributes = ['default-selected-id', 'selected-id', 'tablist-class-name', 'tabs-aria-label'];
+/**
+ * Parses custom web components that are `ds-tab-panel` elements and converts them into React components.
+ * This is necessary for the `Tabs` interface, which expects `TabPanel` children to be React components.
+ * The function handles `ds-tab-panel` elements by converting their kebab-cased attributes to camelCasing,
+ * which is the convention React expects for props.
+ */
+function parseChildren(node) {
+  const elements = findElementsOfType(['ds-tab-panel'], node);
 
-const Wrapper = ({ tabsAriaLabel, ...props }) => {
-  function parseChildren(node) {
-    const elements = findElementsOfType(['ds-tab-panel'], node);
+  return elements.map((element) => {
+    const { children, ...attrs } = element.props || {};
 
-    return elements.map((element) => {
-      const { children, ...attrs } = element.props || {};
-      return createElement(TabPanel, { ...attrs }, children);
-    });
-  }
+    // Rename `root-id` to `id` if it exists.
+    if ('root-id' in attrs) {
+      attrs.id = attrs['root-id'];
+      delete attrs['root-id'];
+    }
+
+    /**
+     * Convert kebab-cased keys to camelCase.
+     * E.g., `tab-class-name` becomes `tabClassName`.
+     */
+    const camelCaseAttrs = Object.keys(attrs).reduce((acc, key) => {
+      const camelCaseKey = key.replace(/-([a-z])/g, (_, letter) => letter.toUpperCase());
+      acc[camelCaseKey] = attrs[key];
+      return acc;
+    }, {} as Partial<TabPanelProps>);
 
+    return createElement(TabPanel, camelCaseAttrs as TabPanelProps, children);
```

Comment:
I think this is a super valid point. The reason I thought keeping the `ds-tab-panel` module was reasonable is that it might help explicitly define the structure and properties of the children that `ds-tabs` can support.

Reply:
I think having structure and guidance on the expected contents of the tabs element is super helpful. Also, I might not be understanding this fully but we need some sort of element called `ds-tab-panel` to have children that populate the tab panel, right? And to do that I think you'll have to register a custom element.

I think as the code is written now, that any HTML that is not contained within a `ds-tab-panel` element will be discarded, right?

Reply:
I think if you don't register `ds-tab-panel` it'll just behave like a `div`. From my quick testing in Chrome, using an unregistered web component doesn't result in errors or warnings.

Reply:
So I guess is it more confusing to have an unregistered custom element serving as a wrapper for content or have additional dead code somewhere (does it end up in the template element?) because the Tabs web component creates the TabPanel component from it's children thus the TabPanel component does not need to be specified separately?

I think I would vote for adding a new class name to handle this distinction. Anything with a class name of ds-tab-panel gets treated as a tab panel, then we can scrap the TabsPanel web component.

Reply:
Thanks @jack-ryan-nava-pbc and @pwolfert  for the helpful comments! I like Jack's idea of using a `ds-tab-panel` class to identify tab panel elements. It seems like it could give us the clarity we want without needing to register a separate ds-tab-panel component. I’m not sure I see any downsides, but I’d be interested to hear from folks if there are potential challenges with it.